assets/js/base/hooks/use-store-events.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { doAction } from '@wordpress/hooks';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Abstraction on top of @wordpress/hooks for dispatching events via doAction for 3rd parties to hook into.
+ */
+export const useStoreEvents = (
+	namespace = 'experimental__woocommerce_blocks'
+): {
+	dispatchStoreEvent: (
+		eventName: string,
+		eventParams: Partial< Record< string, unknown > >
+	) => void;
+} => {
+	const dispatchStoreEvent = useCallback(
+		(
+			eventName: string,
+			eventParams: Partial< Record< string, unknown > >
+		) => {
+			doAction( `${ namespace }-${ eventName }`, eventParams );
```

Comment:
I think we need to wrap this somehow in a try-catch error boundary, or have some control over the actions triggered here.
Throwing a code in `assets/js/extensions/google-analytics/index.ts` for example would break the whole block.

Reply:
Whilst that is true, is our intention to **always** wrap action and filter hooks in try catch? If so, let's abstract it further in a follow up, noting our code is the only consumer of these experimental hooks at present so there won't be breakage we don't know about.

Reply:
isn't the goal to have those hooks public eventually? I don't mind handling this in a follow up as long as we handle it before releasing them, even in experimental. While wrapping SlotFills and CheckoutFilters we ended up changing the implementation to accommodate for try-catch (like limiting filters to strings only).

Reply:
I've added a try catch around the action call so it's isolated.